packages/react-vega/src/Vega.jsx

```diff
@@ -160,21 +154,18 @@ class Vega extends React.Component {
     }
   }
 
-  propsToEmbedOption(props, embedOption) {
-    const embedOptionClone = Object.assign({}, embedOption);
-    ['renderer', 'logLevel', 'tooltip', 'width', 'height', 'padding', 'enableHover']
-      .filter(field => isDefined(props[field]))
-      .forEach(field => {
-        if (field === 'enableHover') {
-          // Since declared type of react vega is enableHover while
-          // vega-embed use hover
-          embedOptionClone.hover = props[field];
-        } else {
-          embedOptionClone[field] = props[field];
-        }
-      });
-
-    return embedOptionClone;
+  propsToEmbedOption(props) {
+    const options = {
+      enableHover: props.enableHover ? props.enableHover : true,
```

Comment:
Change this to `...props.enableHover ? {enableHover: props.enableHover},` as I suggested. The way you wrote it, you have to repeat the defaults.

Reply:
`    const options = {
      ...props.enableHover,
      ...props.height,
      ...props.logLevel,
      ...props.padding,
      ...props.renderer,
      ...props.tooltip,
      ...props.width,
    };`

Since these props we ensure it to have the default value, can I just directly spread it without the conditional?

Reply:
But you are repeating the defaults and if they change in embed, you also have to change them here, no? This could be the right thing if we want to keep different defaults in react vega. Do you think we want that?